### ISSUE TITLE
uncommitted tag messes with remote builds

### ISSUE THREAD

**Issue by swombat** [basecamp/kamal#775]
It's not at all obvious that if you have uncommitted changes while waiting several minutes for the build to finish (which is pretty likely), then Kamal will tag the build with a random string in `configuration.rb`:

```
          [ Kamal::Git.revision, Kamal::Git.uncommitted_changes.present? ? "_uncommitted_#{SecureRandom.hex(8)}" : "" ].join
```

I have no idea what this achieves - perhaps something to help with CI - but it makes deployment more painful because it means I either have to dev in a separate repo on my own machine, or down tools while Kamal does its thing. Otherwise, the delivered build has a tag that `kamal deploy -P` can't find (it generates another new unique tag).

I would suggest making this a configurable option, starting with defaulting to the current behaviour (so as not to break people's builds) with clear documentation about turning this off unless you need it, and eventually making the default what will be helpful for most people who are setting up Kamal (assuming most people don't need these tags).

**Comment by swombat** [comment]
Added a pull request... not sure if this should have been done here instead. Apologies if so.

https://github.com/basecamp/kamal/pull/776

If you want to use that functionality in the meantime you can use the fork at https://github.com/swombat/kamal . See the pull request for simple example of how to use this feature.

**Comment by morgoth** [comment]
@swombat please take a look at https://github.com/basecamp/kamal/pull/700 if this fixes your issue

**Comment by swombat** [comment]
@morgoth Yes, actually that would fix my issue! The point is to make it not default to creating remote builds that can't be accessed, and your solution achieves that.

**Comment by morgoth** [comment]
@swombat can you close the issue then?
